### PR TITLE
Add backport PR Github Action

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Add a github Action to automatically backport PR with `backport/` label

### Motivation

Make backport PR less painful

### CODE REVIEW

File: .github/workflows/backport-pr.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Backport PR
+on:
+  pull_request_target:
+    types:
+      - closed
+      - labeled
```
Comment (amenasria) [inline .github/workflows/backport-pr.yml:6]:
```suggestion

```

**💬 suggestion:** ‏Discussed offline: seems too error prone for the label to be triggered on `labeled`. One misclick and your PR gets backported

File: .github/workflows/backport-pr.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  backport:
+    name: Backport PR
+    runs-on: ubuntu-latest
+    if: >
+      github.event.pull_request.merged
+      && (
+        github.event.action == 'closed'
```
Comment (amenasria) [inline .github/workflows/backport-pr.yml:15]: **💬 suggestion:** ‏This is not needed anymore if we remove `labeled`

File: docs/dev/contributing.md [hunk 3]
```diff
@@ -290,6 +290,7 @@ labels that can be use:
   Python version shipped in the agent).
 - `need-change/operator`, `need-change/helm`: indicate that the configuration needs to be modified in the operator / helm chart as well.
 - `k8s/<min-version>`: indicate the lowest Kubernetes version compatible with the PR's feature.
+- `backport/<branch-name>`: Add this label to have your changes automatically backported to <branch-name>
```
Comment (amenasria) [inline docs/dev/contributing.md:293]:
```suggestion
- `backport/<branch-name>`: Add this label to have your changes automatically backported to `<branch-name>`
```

**💬 suggestion:** ‏It's not displayed well without the backticks:
image

**Review by amenasria — APPROVED** [review]
LGTM !


**👏 praise:** Thanks for this, this is very useful !!‏

**Review by drichards-87 — APPROVED** [review]
Added my own minor suggestion to the existing comment and approved the PR for Docs.

**Pull request merged**